### PR TITLE
add soundcloud connector

### PR DESCRIPTION
**Pull request by gutierri**

### CODE REVIEW

**Comment by f1u77y** [comment]
Thanks for your effort! 

As I can understand, `'.playbackSoundBadge .playbackSoundBadge__lightLink'` contains the user name info, which isn't always the artist name, while `'.playbackSoundBadge .playbackSoundBadge__titleLink span[aria-hidden=true]'` contains title and artist. So, I think it's better to split the latter into artist and title when possible and fallback to what are you doing now when not. You could see an implementation in a similar extension [here](https://github.com/web-scrobbler/web-scrobbler/blob/master/src/connectors/soundcloud.js#L66).

Another issue (but it could be just on my side, so I'd like to ask if you encounter it too) is that extension sends something like "soundcloud.com - search everything" when the playback is paused. I'll try to look into that issue anyway.

The third issue is with album art. It's too small (50x50). I've tried to replace "50x50" with "200x200" and "500x500" in a few random art URLs and it worked, so please do it in this connector.

And just a cosmetic issue: you don't need that empty line 8 at the start of the constructor function.

All the other things about this connector seem fine.

Edit: sorry for the delay.

**Comment by gutierri** [comment]
I made the necessary adjustments. I created a method for bringing artist
and track information. See what you think of such a solution.

About the problem of sending "soundcloud.com - search everything", I did
not understand very well. Send to where? The problem I noticed now,
making corrections was that when the page is first loaded, it can't
start the track, it needs to be done manually. I am seeing how to solve.
Do you have this problem?

**Comment by f1u77y** [comment]
> About the problem of sending "soundcloud.com - search everything", I did
not understand very well. Send to where?

To the background script from content script. Nevermind, I cannot reproduce it now.

> The problem I noticed now, making corrections was that when the page is first loaded, it can't start the track, it needs to be done manually. I am seeing how to solve. Do you have this problem?

I don't have the problem as for now.

File: connectors/soundcloud.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        this.artSelector = '.playbackSoundBadge__avatar span.sc-artwork';
+
+        Utils.query('.playControls .playControls__inner').then(elem => this.observe(elem));
+    }
+
+    get playbackStatus() {
+        return Utils.query(this.playButtonSelector).then(elem => elem.classList.contains('playing') ? 'playing' : 'paused');
+    }
+
+    transform_track_to_artist() {
```
Comment (f1u77y) [inline connectors/soundcloud.js]: First of all, I use camelCase for functions and variables in the whole project, so I think it's better to follow the same code style across all of it.
Secondly, I think `transfrom_track_to_artist` isn't a good name for a method that returns `{artist, track}`. I'd consider something like `getArtistTrack()`.

File: manifest.json [hunk 2]
```diff
@@ -51,6 +51,9 @@
     }, {
         "matches": ["*://music.yandex.ru/*", "*://radio.yandex.ru/*"],
         "js": [ "connectors/yandex-music.js" ]
+    }, {
```
Comment (f1u77y) [inline manifest.json]: Sorry, I've changed the extension layout a bit while this PR was pending. Now this info should be in `background/connectors.js`.

**Review by f1u77y — CHANGES_REQUESTED** [review]

**Comment by gutierri** [comment]
I made a push force and the necessary adjustments

**Review by f1u77y — APPROVED** [review]

**Comment by f1u77y** [comment]
It's fine now but please resolve the conflicts.

**Comment by gutierri** [comment]
Conflicts? It's all right here, and according to the github too!


Got it correct?

**Pull request merged**